sympy/utilities/iterables.py

```diff
@@ -523,9 +521,15 @@ def ibin(n, bits=0, str=False):
     ['000', '001', '010', '011', '100', '101', '110', '111']
 
     """
-    if not str:
+    if n < 0:
+        raise ValueError("Negetive numbers are not allowed")
+
+    if n.bitlength() is not None:
```

Comment:
If you wrap everthing in an `if`, how will `ibin(0)` return `0` or `[0]`? If this isn't added as a test, it should be.

How about just seeing `if bits is not None and n.bit_length() > bits: raise error about needing more bits for this number` else continue with the already existing code

Reply:
Done sir.